test/unit/test_model_store.py

```diff
@@ -32,6 +32,30 @@ def test_import_export_history_failed_job():
     _assert_simple_cat_job_imported(imported_history, state='error')
 
 
+def test_import_export_history_hidden_false_with_hidden_dataset():
+    app = _mock_app()
+
+    u, h, d1, d2, j = _setup_simple_cat_job(app)
+    d2.visible = False
+    app.model.session.flush()
+
+    imported_history = _import_export_history(app, h, export_files="copy", include_hidden=False)
+    assert d2.dataset.get_size() > 0
+    assert imported_history.datasets[-1].get_size() == 0
```

Comment:
I think it's important that we also check that the non-hidden datasets are exported/imported.

```suggestion
    assert d1.dataset.get_size() == imported_history.datasets[0].get_size()
    assert imported_history.datasets[1].get_size() == 0
```

This works locally for me, but if `get_size()` could return slightly different numbers after export/import, the first line can simply be `assert imported_history.datasets[0].get_size() > 0`

Reply:
No, changing the size wouldn't be expected, it's not going through the upload that could do newline conversions or similar data modifying things.

Reply:
I was more worried about the possible difference between real file size and space used on disk.